### DISCUSSION TITLE
How to add texture data to the objects?

### DISCUSSION THREAD

**Discussion by kowshik1234** [zauberzeug/nicegui#1892] [Q&A]
### Question

Hi,

I have the following queries.

1. I want to add wooden texture to some boxes that I have created. I used the below code

```
scene.box(width=1.5, height=1,depth=0.2).move(x=0,y=0,z=0).material(color='#aaaaaa')
```
2. How do I import 3D OBJ files INCLUDING their colour data? I am only able to import stl files but is there any API to import the obj files along with it's mtl files as well?

Any help is appreciarted.

Thanks

**Comment by falkoschindler** [comment]
Hi @kowshik1234!

1. That's currently not supported. You need to use individual `scene.texture` objects.
2. That's also not possible at the moment.

All in all we haven't spent much time working on `ui.scene` in the last year. Other elements and features kept us quite busy and the 3D scene was good enough for our applications. But there's certainly room for improvement. Pull requests and other contributions are always welcome. 🙂

**Reply by kowshik1234** [reply]
Well, `scene.texture` doesn't really cut to my needs. Can I know if I can use any external libraries and import it to the nicegui?

**Reply by natankeddem** [reply]
I am not familiar with Javascript based 3D libraries but depending on your library you might be able to make it work on your own with some effort. For example I implemented xterm and posted a demo here: https://github.com/zauberzeug/nicegui/discussions/1846

**Comment by liushuya7** [comment]
Would it be possible to add support for `*.dae` files?

**Reply by falkoschindler** [reply]
I guess that would be possible.

By the way, PR https://github.com/zauberzeug/nicegui/pull/2532 added support for GLTF meshes, which include texture.
